### PR TITLE
Add support for Registration V2 API 

### PR DESCRIPTION
**Pull request by sangeetha5491**
## Description

Add RegistrationCreateModel, RegistrationUpdateModel, EventsOfInterestInputModel
Update Registration model to include HAL links and other changes wrt the new registration API response
Update Jackson Object Mapper to support HAL JSON
Update the integration and unit tests
Other fixes to support these changes

## Related Issue

* GH-131

## Motivation and Context

Adds support for the latest Registration IO Management API. For more details see: https://developer.adobe.com/events/docs/api/#tag/Registrations

## How Has This Been Tested?

Integration tests and unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the [code style](CODE_STYLE.md) of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: events_mgmt/src/main/java/com/adobe/aio/event/management/model/RegistrationCreateModel.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import org.apache.commons.lang3.StringUtils;
+
+@JsonInclude(Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class RegistrationCreateModel extends RegistrationUpdateModel {
+
+    @JsonProperty("client_id")
+    private final String clientId;
+
+    public RegistrationCreateModel(final String clientId, final String name, final String description,
```
Comment (francoisledroff) [inline events_mgmt/src/main/java/com/adobe/aio/event/management/model/RegistrationCreateModel.java:30]: same here, let's make this private

File: events_mgmt/src/main/java/com/adobe/aio/event/management/model/Registration.java [hunk 2]
```diff
@@ -11,65 +11,131 @@
...
-    VERIFIED
-  }
+  @Link
+  private HALLink self;
 
-  public enum IntegrationStatus {
-    ENABLED, DISABLED
-  }
+  @Link("rel:events")
+  private HALLink eventsUrl;
```
Comment (francoisledroff) [inline events_mgmt/src/main/java/com/adobe/aio/event/management/model/Registration.java:34]: why not keeping it as `journalUrl` ?

File: events_mgmt/src/main/java/com/adobe/aio/event/management/model/EventsOfInterestInputModel.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+public class EventsOfInterestInputModel {
+
+  @JsonProperty("event_code")
+  String eventCode;
+
+  @JsonProperty("provider_id")
+  String providerId;
+
+  private EventsOfInterestInputModel() {
```
Comment (francoisledroff) [inline events_mgmt/src/main/java/com/adobe/aio/event/management/model/EventsOfInterestInputModel.java:27]: this is unused , right ?

File: core/src/main/java/com/adobe/aio/util/JacksonUtil.java [hunk 4]
```diff
@@ -24,9 +32,15 @@ private JacksonUtil() {
...
 
   public static final ObjectMapper DEFAULT_OBJECT_MAPPER =
-      new ObjectMapper()
-          .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
-          .findAndRegisterModules();
+    JsonMapper.builder()
+    .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    .serializationInclusion(Include.NON_NULL)
+    .addModule(new SimpleModule()
+      .addSerializer(ZonedDateTime.class, new ZonedDateTimeSerializer())
```
[the remark this answers is not in the document]
Comment (sangeetha5491) [reply core/src/main/java/com/adobe/aio/util/JacksonUtil.java:39]: need to check.

File: core/pom.xml [hunk 5]
```diff
@@ -54,6 +54,14 @@
...
       <artifactId>feign-form</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.openapitools.jackson.dataformat</groupId>
+      <artifactId>jackson-dataformat-hal</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.cloudevents</groupId>
+      <artifactId>cloudevents-api</artifactId>
+    </dependency>
```
Comment (francoisledroff) [inline core/pom.xml:64]: I guess you could exclude the `org.hibernate.validator:hibernate-validator` as a new  `<exclusions><exclusion>` entry here

**Review by francoisledroff — COMMENTED** [review]
looking good !

**Review by francoisledroff — APPROVED** [review]

**Pull request merged**